Modules/Messages/test/Messages.Application.Tests/MessagesRepositoryTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.Persistence.BlobStorage;
+using Backbone.BuildingBlocks.Application.Pagination;
+using Backbone.Modules.Devices.Application.Tests;
+using Backbone.Modules.Messages.Application.Infrastructure.Persistence;
+using Backbone.Modules.Messages.Domain.Entities;
+using Backbone.Modules.Messages.Domain.Ids;
+using Backbone.Modules.Messages.Infrastructure.Persistence.Database;
+using Backbone.Modules.Messages.Infrastructure.Persistence.Database.Repository;
+using Backbone.Tooling;
+using Backbone.UnitTestTools.TestDoubles.Fakes;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Backbone.Modules.Messages.Application.Tests;
+
+public class MessagesRepositoryTest
+{
+    private readonly MessagesDbContext _arrangeContext;
+    private readonly MessagesDbContext _actContext;
+
+    public MessagesRepositoryTest()
+    {
+        (_arrangeContext, _actContext, _) = FakeDbContextFactory.CreateDbContexts<MessagesDbContext>();
+    }
+
+    [Fact]
+    public async Task Message_body_can_be_retrieved_from_blob_storage_if_unavailable_in_database()
+    {
+        // Arrange
+        var identityAddress = TestDataGenerator.CreateRandomIdentityAddress();
+        var deviceId = TestDataGenerator.CreateRandomDeviceId();
+
+        var messageWithFilledBody = new Message(
+            identityAddress,
+            deviceId,
+            SystemTime.UtcNow,
+            new byte[] { 1, 2, 3 },
```

Comment:
I would simplify this to `new byte[] { 1 }`. Then, in the blob storage, you can use `new byte[] { 2 }`.